### PR TITLE
fix: task validation error when adding tasks to projects

### PR DESCRIPTION
**Pull request by benknowles**
When adding a task to a project, if the project didn't have an Expected End Date the validation would fail.

This is because passing a None value to getdate() returns today's date, rather than being optional as expected.

### CODE REVIEW

File: erpnext/projects/doctype/task/task.py [hunk 1]
```diff
@@ -47,9 +47,10 @@ def validate_parent_project_dates(self):
 		if not self.project or frappe.flags.in_test:
 			return
 
-		expected_end_date = getdate(frappe.db.get_value("Project", self.project, "expected_end_date"))
+		expected_end_date = frappe.db.get_value("Project", self.project, "expected_end_date")
 
 		if expected_end_date:
+			expected_end_date = getdate(expected_end_date)
```
Comment (hrwX) [inline erpnext/projects/doctype/task/task.py]: Pass the argument as below
```suggestion
			validate_project_dates(getdate(expected_end_date), self, "act_start_date", "act_end_date", "Actual")
```

File: erpnext/projects/doctype/task/task.py [hunk 2]
```diff
@@ -47,9 +47,10 @@ def validate_parent_project_dates(self):
...
 			validate_project_dates(expected_end_date, self, "exp_start_date", "exp_end_date", "Expected")
```
Comment (hrwX) [inline erpnext/projects/doctype/task/task.py]: Pass the argument as below
```suggestion
			validate_project_dates(getdate(expected_end_date), self, "act_start_date", "act_end_date", "Actual")
```

**Review by hrwX — CHANGES_REQUESTED** [review]

**Comment by hrwX** [comment]
@benknowles Also open a new PR for the fix for version-12-hotfix branch

File: erpnext/projects/doctype/task/task.py (hunk 2 above)
Comment (Thunderbottom) [reply erpnext/projects/doctype/task/task.py]: there's already a validation for actual task dates below this.

**Review by hrwX — APPROVED** [review]

**Comment by benknowles** [comment]
@rohitwaghchaure Could you merge this PR please, related to #19936 in develop

**Pull request merged**